src/sidebar/service-url.js

```diff
@@ -3,44 +3,91 @@
 var urlUtil = require('./util/url-util');
 
 /**
- * A map of all route names to relative URLs on the Hypothesis service that
- * the client links to.
+ * An AngularJS service that returns an absolute URL given a link name and
```

Comment:
A more generic alternative way of expressing this would be "A factory that returns ...". An AngularJS _service_ is just a class which is instantiated once per application, where the constructor arguments are supplied by the DI system.